core/src/main/java/org/pentaho/reporting/platform/plugin/ParameterXmlContentHandler.java

```diff
@@ -747,7 +752,7 @@ Element createParameterElement( final ParameterDefinitionEntry parameter,
 
         //parameter dependencies: see backlog-7980
         boolean shouldValidateOnServer =
-          !dependencies.isNoDependencyInformationAvailable()
+          !dependencies.getDependencyGraph().isEmpty()
```

Comment:
I tried different tests with the 11 different tickets I am verifying against (BISERVER-14306 BISERVER-14206 BISERVER-14297 BISERVER-14315 BISERVER-14308 PRD-6082 BISERVER-13435 PRD-5982 BACKLOG-26181 BISERVER-14198 BISERVER-14254) and I came to the conclusion that "isNoDependencyInformationAvailable / isDependencyInformationAvailable / dependencies.getDependencyGraph().isEmpty()" seems to be not helping.  By having one of these checks in, it keeps BACKLOG-26181 from working (parameters are hidden). By having an && check in instead, it keeps the bean shell script from running on PRD-5982 properly (it'll update only once and never again). When I remove that check and only check against "shouldAlwaysValidateOnServer", then all the tickets respond positively to it.  My initial change to !dependencies.getDependencyGraph().isEmpty() was expecting that if any dependencies existed, then it should be validated.  It actually should only be validated if the parameter has a dependency, which is checked in the next conditional statement, so my initial change was unnecessary.

The original code was conflating No Dependencies and Unable to Process Dependencies before, by reversing my logic I'm now stating that you are able to process dependencies successfully (regardless of if there are dependencies or not). So in essence, the old code check was verifying if dependencies existed.  But this is basically covered by checking if a parameter has a dependency in the next conditional statement, which then tells the client code to validate on the server.  So by removing the check, we end up having less redundancy.

There were some issues as well with the old code and using the isDependencyInformationAvailable, similar problems, where if the old one came back overall with true, or if isDependencyInformationAvailable was true, then it would state every parameter should be validated (which is not true for every parameter) and caused some issues with the reporting plugin.  We only need to force validation of parameters that need it, so by removing the redundant and too general boolean, we end up being more accurate (the 11 tickets are satisfied).